### PR TITLE
Add 'unmark' command

### LINKED ISSUE #64: 'Unmark' appointment feature
As a user, I can unmark an appointment as undone so that I can revert marking appointments when I mark them by mistake.

### LINKED ISSUE #65: Add 'unmark' command to mark appointments

### PR DESCRIPTION
**Pull request by NicholasTYD**
Closes #63
Closes #64 
Closes #65 

Build upon 'mark' command. 
Group common methods in mark/unmark classes to reduce need for code duplication. 
Fix jUnit occasionally failing due to interactions between static variables and jUnit.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -36,6 +39,30 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
...
+        String[] indexes = trimmedIndexes.split(indexSeparator);
+
+        List<Index> indexList = new ArrayList<>();
+        for (String index : indexes) {
+            indexList.add(parseIndex(index));
+        }
+
+        if (indexList.size() != parseCount) {
+            throw new ParseException(MESSAGE_UNEXPECTED_INDEX_COUNT);
+        }
```
Comment (JovonLim) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:62]: Would be better to move it upwards before parsing the indexes to make the happy path prominent.

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.Appointment;
+import seedu.address.model.person.Person;
+import seedu.address.testutil.PersonBuilder;
+
+class MarkCommandTest {
```
Comment (JovonLim) [inline src/test/java/seedu/address/logic/commands/MarkCommandTest.java:23]: Perhaps could include some tests for the markCommandParser as well. Similarly for the unmarkCommandParser

**Review by JovonLim — APPROVED** [review]
LGTM overall

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (NicholasTYD) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:62]: That's a nice suggestion

File: src/test/java/seedu/address/logic/commands/MarkCommandTest.java (hunk 2 above)
Comment (NicholasTYD) [reply src/test/java/seedu/address/logic/commands/MarkCommandTest.java:23]: That's a good suggestion

**Review by pwjj2000 — APPROVED** [review]
LGTM!

**Pull request merged**